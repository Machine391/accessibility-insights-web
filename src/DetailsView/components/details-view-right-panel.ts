// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { DetailsViewRightContentPanelType } from 'common/types/store-data/details-view-right-content-panel-type';
import { ReactFCWithDisplayName } from '../../common/react/named-fc';
import { DetailsViewPivotType } from '../../common/types/store-data/details-view-pivot-type';
import {
    getOverviewTitle,
    getTestViewTitle,
    GetTestViewTitleProps,
} from '../handlers/get-document-title';
import {
    GetLeftNavSelectedKeyProps,
    getOverviewKey,
    getTestViewKey,
} from './left-nav/get-left-nav-selected-key';
import {
    OverviewContainer,
    OverviewContainerDeps,
    OverviewContainerProps,
} from './overview-content/overview-content-container';
import { TargetChangeDialogDeps } from './target-change-dialog';
import {
    TestViewContainer,
    TestViewContainerDeps,
    TestViewContainerProps,
} from './test-view-container';

export type DetailsViewContentDeps = OverviewContainerDeps &
    TestViewContainerDeps &
    TargetChangeDialogDeps;

export type RightPanelProps = Omit<TestViewContainerProps, 'deps'> &
    Omit<OverviewContainerProps, 'deps'> & {
        deps: OverviewContainerDeps | TestViewContainerDeps;
    };

export type DetailsRightPanelConfiguration = Readonly<{
    RightPanel: ReactFCWithDisplayName<RightPanelProps>;
    GetTitle: (props: GetTestViewTitleProps) => string;
    GetLeftNavSelectedKey: (props: GetLeftNavSelectedKeyProps) => string;
    GetStartOverContextualMenuItemKeys: () => string[];
}>;

export type GetDetailsRightPanelConfiguration = (
    props: GetDetailsRightPanelConfigurationProps,
) => DetailsRightPanelConfiguration;
export type GetDetailsRightPanelConfigurationProps = {
    selectedDetailsViewPivot: DetailsViewPivotType;
    detailsViewRightContentPanel: DetailsViewRightContentPanelType;
};

const TestViewPanel: DetailsRightPanelConfiguration = {
    RightPanel: TestViewContainer,
    GetTitle: getTestViewTitle,
    GetLeftNavSelectedKey: getTestViewKey,
    GetStartOverContextualMenuItemKeys: () => ['assessment', 'test'],
};

const detailsViewOverviewConfiguration: {
    [key in DetailsViewPivotType]: DetailsRightPanelConfiguration;
} = {
    [DetailsViewPivotType.assessment]: {
        RightPanel: OverviewContainer,
        GetTitle: getOverviewTitle,
        GetLeftNavSelectedKey: getOverviewKey,
        GetStartOverContextualMenuItemKeys: () => ['assessment'],
    },
    [DetailsViewPivotType.mediumPass]: {
        RightPanel: OverviewContainer,
        GetTitle: getOverviewTitle,
        GetLeftNavSelectedKey: getOverviewKey,
        GetStartOverContextualMenuItemKeys: () => ['assessment'],
    },
    [DetailsViewPivotType.fastPass]: TestViewPanel,
};

export const GetDetailsRightPanelConfiguration: GetDetailsRightPanelConfiguration = (
    props: GetDetailsRightPanelConfigurationProps,
) => {
    if (props.detailsViewRightContentPanel === 'TestView') {
        return TestViewPanel;
    }
    return detailsViewOverviewConfiguration[props.selectedDetailsViewPivot];
};
