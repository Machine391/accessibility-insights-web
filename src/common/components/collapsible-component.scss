// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

span.collapsible-title {
    padding-left: 0.5vw;
}

.collapsible {
    display: inline-flex;
    font-size: 26px;
    height: auto;
    width: auto;
    text-align: left;
    color: #106ebe;

    h2 {
        font-weight: normal;
        font-size: 17px;
    }
}

.collapsible-content {
    overflow: hidden;
    padding-left: calc(0.5vw + 13px); // 13px is Chevron width.
}
