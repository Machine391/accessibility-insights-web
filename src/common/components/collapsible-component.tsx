// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ActionButton } from '@fluentui/react';
import { Button, themeToTokensObject, webLightTheme } from '@fluentui/react-components';
import { css } from '@fluentui/utilities';
import styles from 'common/components/collapsible-component.scss';
import * as React from 'react';
import { ChevronDown24Regular, ChevronRight24Regular } from '@fluentui/react-icons';

export interface CollapsibleComponentProps {
    header: JSX.Element;
    content: JSX.Element;
    contentClassName?: string;
    containerClassName?: string;
}

interface CollapsibleComponentState {
    showContent: boolean;
}

export class CollapsibleComponent extends React.Component<
    CollapsibleComponentProps,
    CollapsibleComponentState
> {
    private readonly iconNameDown = 'ChevronDown';
    private readonly iconNameUp = 'ChevronRight';

    constructor(props: CollapsibleComponentProps) {
        super(props);
        this.state = { showContent: true };
    }

    private onClick = (): void => {
        this.setState(prevState => ({ showContent: !prevState.showContent }));
    };

    public render(): JSX.Element {
        const showContent = this.state.showContent;
        let iconName = this.iconNameUp;
        let content: JSX.Element | null = null;

        if (showContent) {
            iconName = this.iconNameDown;
            content = (
                <div className={css(this.props.contentClassName, styles.collapsibleContent)}>
                    {this.props.content}
                </div>
            );
        }

        return (
            <div className={css(this.props.containerClassName)}>
                {/* 111<ActionButton
                    className={styles.collapsible}
                    iconProps={{ iconName: iconName }}
                    onClick={this.onClick}
                    aria-expanded={showContent}
                >
                    {this.props.header}
                </ActionButton> */}
                <Button className={styles.collapsible} appearance='transparent'
                    onClick={this.onClick}
                    aria-expanded={showContent}>
                    {showContent ? (
                        <ChevronDown24Regular color={themeToTokensObject(webLightTheme).colorCompoundBrandStrokeHover} className={styles.collapsible} />
                    ) : (
                        <ChevronRight24Regular color={themeToTokensObject(webLightTheme).colorCompoundBrandStrokeHover} className={styles.collapsible} />
                    )}
                    111
                    {this.props.header}
                </Button>
                {content}
            </div>
        );
    }
}
