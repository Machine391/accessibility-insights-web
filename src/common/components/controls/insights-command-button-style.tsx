// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { makeStyles, tokens } from '@fluentui/react-components';

export const InsightsCommandButtonStyle = makeStyles({
    button: { fontWeight: tokens.fontWeightRegular },
    buttonIcon: {
        color: tokens.colorCompoundBrandStrokeHover,
    },
});
