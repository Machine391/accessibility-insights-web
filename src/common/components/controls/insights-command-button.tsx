// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { css } from '@fluentui/react';
import { Button, ButtonProps } from '@fluentui/react-components';
import { Icons } from 'common/icons/fluentui-v9-icons';
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';
import styles from './insights-command-button.scss';

React.useEffect;
export type InsightsCommandButtonProps = ButtonProps & {
    iconName?: any;
    componentRef?: any;
    text?: any;
    menuIconProps?: any;
};

// See https://www.figma.com/file/Wj4Ggf6GGQBQkiDIaHfXRX2B/Accessibility-Insights%3A-Styles?node-id=1%3A27

export const InsightsCommandButton = NamedFC<InsightsCommandButtonProps>(
    'InsightsCommandButton',
    props => {
        const iconComponent = props.iconName;
        const menuIconComponent = props.menuIconProps ? props.menuIconProps.iconName : null;

        return (
            <>
                <h3>InsightsCommandButton</h3>
                <Button
                    appearance="transparent"
                    {...props}
                    icon={Icons[iconComponent]}
                    className={css(styles.insightsCommandButton, props.className)}
                    ref={props.componentRef}
                >
                    {props.text}
                    {menuIconComponent !== null && Icons[menuIconComponent]}
                </Button>
            </>
        );
    },
);
