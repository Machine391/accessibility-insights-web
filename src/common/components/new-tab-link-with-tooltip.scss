// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';

:global(.high-contrast-theme) {
    .insights-tooltip {
        outline-style: solid;
        outline-width: 1px;
        outline-color: $always-white;
    }
}

.insights-tooltip-host {
    display: inline-block;
    min-width: 0;
}

.insights-link {
    display: flex;

    &:hover,
    &:active {
        text-decoration: none !important;
    }

    &:focus-visible {
        box-shadow: $neutral-60 0 0 0 1px inset !important;
        outline: $neutral-60 auto 1px !important;
        text-decoration: none !important;
    }
}
