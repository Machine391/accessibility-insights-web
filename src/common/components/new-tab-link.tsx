// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Link, LinkProps, mergeClasses } from '@fluentui/react-components';
import * as React from 'react';

import { NamedFC } from '../react/named-fc';

export type NewTabLinkProps = LinkProps & { as?: 'a' };

export const NewTabLink = NamedFC<NewTabLinkProps>('NewTabLink', ({ className, ...props }) => {
    return <Link className={mergeClasses('insights-link', className)} target="_blank" {...props} />;
});
