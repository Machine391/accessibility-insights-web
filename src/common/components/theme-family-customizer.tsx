// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ThemeProvider, PartialTheme } from '@fluentui/react';
import { FluentProvider, webDarkTheme, webLightTheme } from '@fluentui/react-components';
import { NamedFC } from 'common/react/named-fc';
import { DefaultTheme } from 'common/styles/default-theme';
import { FastPassTheme } from 'common/styles/fast-pass-theme';
import { FastPassV9Theme } from 'common/styles/fast-pass-v9-theme';
import { HighContrastTheme } from 'common/styles/high-contrast-theme';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
import * as React from 'react';

export type ThemeFamily = 'default' | 'fast-pass';

const themeFamilyDefaultThemes: { [themeFamily in ThemeFamily]: PartialTheme } = {
    default: DefaultTheme,
    'fast-pass': FastPassTheme,
};

const themeFamilyDefaultV9Themes: { [themeFamily in ThemeFamily] } = {
    default: webLightTheme,
    'fast-pass': FastPassV9Theme,
};

export type ThemeFamilyCustomizerProps = {
    userConfigurationStoreData: UserConfigurationStoreData;
    themeFamily: ThemeFamily;
    children: React.ReactNode;
};

// This is a store-state-aware version of the Fluent UI <ThemeProvider> component. It's intended
// as a wrapper for Fluent UI-based components that need to use a different Theme from our default
// one, but still want to support being overriden by HighContrastTheme when a user has enabled our
// High Contrast Mode setting (or a native system High Contrast setting).
export const ThemeFamilyCustomizer = NamedFC<ThemeFamilyCustomizerProps>(
    'ThemeCustomizer',
    props => {
        const isHighContrastEnabled = props.userConfigurationStoreData?.enableHighContrast === true;
        const defaultTheme = themeFamilyDefaultThemes[props.themeFamily];
        const defaultV9Theme = themeFamilyDefaultV9Themes[props.themeFamily]
        const activeTheme = isHighContrastEnabled ? HighContrastTheme : defaultTheme;
        const activeV9Theme = isHighContrastEnabled ? { ...webDarkTheme, colorNeutralBackground1: '#161616 ' } : defaultV9Theme;
        return <ThemeProvider applyTo="body" theme={activeTheme}><FluentProvider theme={activeV9Theme}>{props.children}</FluentProvider></ThemeProvider>;
    },
);
