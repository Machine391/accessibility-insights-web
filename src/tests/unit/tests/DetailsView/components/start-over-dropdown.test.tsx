// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ContextualMenu, IButton, IRefObject } from '@fluentui/react';
import { Menu, MenuItem, MenuList, MenuPopover, MenuTrigger } from '@fluentui/react-components';
import { act, fireEvent, render, screen } from '@testing-library/react';

import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
import { StartOverContextMenuKeyOptions } from 'DetailsView/components/details-view-right-panel';
import { StartOverDialogType } from 'DetailsView/components/start-over-dialog';
import * as React from 'react';
import {
    expectMockedComponentPropsToMatchSnapshots,
    getMockComponentClassPropsForCall,
    mockReactComponents,
    useOriginalReactElements,
} from 'tests/unit/mock-helpers/mock-module-helpers';
import { IMock, Mock, Times } from 'typemoq';

import {
    StartOverDropdown,
    StartOverProps,
} from '../../../../../DetailsView/components/start-over-dropdown';
jest.mock('@fluentui/react');
jest.mock('@fluentui/react-components');
jest.mock('common/components/controls/insights-command-button');

describe('StartOverDropdownTest', () => {
    mockReactComponents([Menu, MenuTrigger, MenuPopover, MenuList, MenuItem]);
    let defaultProps: StartOverProps;
    let openDialogMock: IMock<(dialogType: StartOverDialogType) => void>;

    beforeEach(() => {
        openDialogMock = Mock.ofInstance(() => null);
        defaultProps = {
            singleTestSuffix: 'single test suffix stub',
            allTestSuffix: 'all test suffix stub',
            rightPanelOptions: { showTest: true },
            switcherStartOverPreferences: { showTest: true },
            dropdownDirection: 'down',
            openDialog: openDialogMock.object,
            buttonRef: {} as IRefObject<IButton>,
            hasSubMenu: true
        };
    });

    it('render', () => {
        const renderResult = render(<StartOverDropdown {...defaultProps} />);

        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Menu, MenuTrigger, MenuPopover, MenuList, MenuItem]);
    });

    it('render ContextualMenu', async () => {
        useOriginalReactElements('@fluentui/react-components', ['Menu', 'MenuTrigger', 'MenuPopover', 'MenuList', 'MenuItem'])
        render(<StartOverDropdown {...defaultProps} />);

        const subMenu = screen.getByRole('menuitem');
        fireEvent.click(subMenu);

        const startAll = screen.getByText('Start over all test suffix stub');
        const startSingle = screen.getByText('Start over single test suffix stub');
        expect(startAll).toBeDefined();
        expect(startSingle).toBeDefined();
        // await act(() =>
        //     getMockComponentClassPropsForCall(InsightsCommandButton).onClick({
        //         currentTarget: 'test event',
        //     }),
        // );
        // expect(renderResult.asFragment()).toMatchSnapshot();

        // const mockProps = getMockComponentClassPropsForCall(ContextualMenu);
        //expect(mockProps.target).toBe('test event');
    });

    const menuButtonOptions = [true, false];
    const optionName = 'showTest';
    const optionKey = 'test';
    menuButtonOptions.forEach(rightPanelOptionEnabled => {
        menuButtonOptions.forEach(switcherPreferencesOptionEnabled => {
            const rightPanelOptions = {
                [optionName]: rightPanelOptionEnabled,
            } as StartOverContextMenuKeyOptions;
            const switcherPreferences = {
                [optionName]: switcherPreferencesOptionEnabled,
            } as StartOverContextMenuKeyOptions;

            const shouldFindOption = rightPanelOptionEnabled && switcherPreferencesOptionEnabled;

            const casePrefix = shouldFindOption
                ? `${optionKey} item IS rendered`
                : `${optionKey} item IS NOT rendered`;

            // test(`${casePrefix} - rightPanelOptions.${optionName} is ${rightPanelOptionEnabled} & switcherStartOverPreferences.${optionName} is ${switcherPreferencesOptionEnabled}`, async () => {
            //     defaultProps.rightPanelOptions = rightPanelOptions;
            //     defaultProps.switcherStartOverPreferences = switcherPreferences;

            //     render(<StartOverDropdown {...defaultProps} />);
            //     await act(() =>
            //         getMockComponentClassPropsForCall(InsightsCommandButton).onClick({
            //             currentTarget: 'test target',
            //         }),
            //     );

            //     const isStartOverOptionRendered = getMockComponentClassPropsForCall(
            //         ContextualMenu,
            //     ).items.some(item => item.key === optionKey);
            //     expect(isStartOverOptionRendered).toEqual(shouldFindOption);
            // });
        });
    });

    it('render with dropdown on left', async () => {
        const props: StartOverProps = {
            ...defaultProps,
            dropdownDirection: 'left',
        };

        const renderResult = render(<StartOverDropdown {...props} />);
        const subMenu = screen.getByRole('menuitem');
        fireEvent.click(subMenu);
        // await act(() =>
        //     getMockComponentClassPropsForCall(InsightsCommandButton).onClick({
        //         currentTarget: 'test target',
        //     }),
        // );
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it('should open the start test over dialog', async () => {
        useOriginalReactElements('@fluentui/react-components', ['MenuPopover', 'MenuList', 'MenuItem'])
        openDialogMock.setup(sds => sds('test')).verifiable(Times.once());


        const renderResult = render(<StartOverDropdown {...defaultProps} />);
        //console.log('test===>', getMockComponentClassPropsForCall(MenuuItem))
        //getMockComponentClassPropsForCall(MenuItem).onClick;
        // await act(() =>
        //     getMockComponentClassPropsForCall(MenuItem).onClick({
        //         currentTarget: 'test target',
        //     }),
        // );

        const subMenu = screen.getByRole('menuitem');
        fireEvent.click(subMenu);
        // const subMenuValue = screen.getByText('Start over all test suffix stub')
        // console.log('subMenuValue', subMenuValue)
        //fireEvent.click(subMenuValue);

        // await act(() =>
        //     getMockComponentClassPropsForCall(MenuItem).onClick({
        //         currentTarget: 'test target',
        //     }),
        // );


        // getMockComponentClassPropsForCall(MenuItem)
        //     .items.find(elem => elem.key === 'test')
        //     .onClick();


        // await act(() =>
        //     getMockComponentClassPropsForCall(MenuItem)?.onClick({})
        // );
        //console.log('here---->', getMockComponentClassPropsForCall(MenuItem))
        // const menuValue = screen.getByRole('menuitem');
        // fireEvent.click(menuValue);
        //
        // await act(() =>
        //     getMockComponentClassPropsForCall(Menu).onClick({
        //         currentTarget: 'test target',
        //     }),
        // );
        // await act(() =>
        //     getMockComponentClassPropsForCall(MenuItem)
        //         .items.find(elem => elem.key === 'test')
        //         .onClick()
        // );
        renderResult.debug();
        openDialogMock.verifyAll();
    });

    // it('should open the start assessment over dialog', async () => {
    //     openDialogMock.setup(sds => sds('assessment')).verifiable(Times.once());

    //     render(<StartOverDropdown {...defaultProps} />);
    //     await act(() =>
    //         getMockComponentClassPropsForCall(Menu).onClick({
    //             currentTarget: 'test target',
    //         }),
    //     );
    //     getMockComponentClassPropsForCall(ContextualMenu)
    //         .items.find(elem => elem.key === 'assessment')
    //         .onClick();

    //     openDialogMock.verifyAll();
    // });

    // it('should dismiss the contextMenu', async () => {
    //     const renderResult = render(<StartOverDropdown {...defaultProps} />);
    //     await act(() =>
    //         getMockComponentClassPropsForCall(Menu).onClick({
    //             currentTarget: 'test target',
    //         }),
    //     );
    //     await act(() => getMockComponentClassPropsForCall(ContextualMenu).onDismiss());
    //     const mockContextualmenu = renderResult.container.querySelector('mock-contextualmenu');
    //     expect(mockContextualmenu).toBeNull();
    // });
});
