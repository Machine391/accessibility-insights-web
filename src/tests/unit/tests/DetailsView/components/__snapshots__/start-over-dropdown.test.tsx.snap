// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartOverDropdownTest render 1`] = `
<DocumentFragment>
  <div>
    <mock-menu>
      <mock-menutrigger
        disablebuttonenhancement="true"
      >
        <mock-menubutton
          appearance="transparent"
          aria-label="start over menu"
          icon="[object Object]"
          menuicon="[object Object]"
        >
          Start over V9
        </mock-menubutton>
      </mock-menutrigger>
      <mock-menupopover
        style="border-radius: unset !important;"
      >
        <mock-menulist>
          <mock-menuitem
            name="Start over all test suffix stub"
          >
            Start over all test suffix stub
          </mock-menuitem>
          <mock-menuitem
            name="Start over single test suffix stub"
          >
            Start over single test suffix stub
          </mock-menuitem>
        </mock-menulist>
      </mock-menupopover>
    </mock-menu>
  </div>
</DocumentFragment>
`;

exports[`StartOverDropdownTest render with dropdown on left 1`] = `
<DocumentFragment>
  <div>
    <mock-menu>
      <mock-menutrigger
        disablebuttonenhancement="true"
      >
        <mock-menubutton
          appearance="transparent"
          aria-label="start over menu"
          icon="[object Object]"
          menuicon="[object Object]"
        >
          Start over V9
        </mock-menubutton>
      </mock-menutrigger>
      <mock-menupopover
        style="border-radius: unset !important;"
      >
        <mock-menulist>
          <mock-menuitem
            name="Start over all test suffix stub"
          >
            Start over all test suffix stub
          </mock-menuitem>
          <mock-menuitem
            name="Start over single test suffix stub"
          >
            Start over single test suffix stub
          </mock-menuitem>
        </mock-menulist>
      </mock-menupopover>
    </mock-menu>
  </div>
</DocumentFragment>
`;

exports[`StartOverDropdownTest render with dropdown on left: MenuButton props 1`] = `
{
  "appearance": "transparent",
  "aria-label": "start over menu",
  "children": "Start over V9",
  "icon": <ForwardRef(ArrowClockwiseRegular)
    className="refreshIcon"
  />,
  "menuIcon": <span
    className="chevron"
  >
    <ForwardRef(ChevronRight20Regular) />
  </span>,
}
`;

exports[`StartOverDropdownTest render: MenuButton props 1`] = `
{
  "appearance": "transparent",
  "aria-label": "start over menu",
  "children": "Start over V9",
  "icon": <ForwardRef(ArrowClockwiseRegular)
    className="refreshIcon"
  />,
  "menuIcon": <span
    className="chevron"
  >
    <ForwardRef(ChevronDown20Regular) />
  </span>,
}
`;

exports[`StartOverDropdownTest render: MenuList props 1`] = `
{
  "children": [
    <ForwardRef(MenuItem)
      name="Start over all test suffix stub"
      onClick={[Function]}
    >
      Start over all test suffix stub
    </ForwardRef(MenuItem)>,
    <ForwardRef(MenuItem)
      name="Start over single test suffix stub"
      onClick={[Function]}
    >
      Start over single test suffix stub
    </ForwardRef(MenuItem)>,
  ],
}
`;
